### DISCUSSION TITLE
Access API (38888) via Websockets (ws://) vs tcp://

### DISCUSSION THREAD

**Discussion by lh32469** [ravendb/ravendb#15891] [Ideas]
Would it be possible to access port 38888 via websockets vs tcp so that connections could be made via http "upgrade" and it would make it easier to route via k8s Ingress?  Much like Mosquito MQTT.

Something like:  
    RAVEN_PublicServerUrl_Tcp=ws://foo.bar/api
or
   RAVEN_PublicServerUrl_Tcp=wss://foo.bar/api

I did a search for websockets and didn't get any hits.

Thanks,

**Comment by georgiosd** [comment]
Hey @lh32469 - there's no good reason to add another layer between Raven and clients - see [this thread](https://github.com/ravendb/ravendb/discussions/15436#discussioncomment-4437055).

**Comment by lh32469** [comment]
Thanks, it was just a thought..

I have a single node RavenDB running inside a k8s cluster and I’m able to access it via http ingress (RAVEN_PublicServerUrl).

I’m trying to get Data Subscriptions to work on a client outside the cluster and the Subscription is created in RavenDB but then the client gets Connection refused which I think is because it’s trying to connect to the tcp port 38888.  I’m using the Java 5.4.0 client and the DB is 5.4.101

I recently setup Mosquitto (https://mosquitto.org) which is an MQTT Broker that can "tunnel" MQTT inside Websockets inside https.  At least that's my understanding of it and that's where my suggestion came from.  

Ultimately, if this works, the App would be running inside the same k8s cluster so direct access won't be an issue and for dev purposes i can do port forwarding via k8s or MetalLB on the cluster itself.

Thanks for the quick response.

**Reply by ayende** [reply]
The typical way to do that in `k8s` is to have _separate hosts_ for Http and Tcp, given the limit of `k8s` to route based on port.

Note that RavenDB needs to do TLS termination, so the actual protocol that is being used is not relevant.
